server/mdm/maintainedapps/apps.json

```diff
@@ -12,7 +12,15 @@
 	{
 		"identifier": "box-drive",
 		"bundle_identifier": "com.box.desktop",
-		"installer_format": "pkg"
+		"installer_format": "pkg",
+		"pre_uninstall_scripts": [
+			"(cd /Users/$LOGGED_IN_USER; sudo -u $LOGGED_IN_USER fileproviderctl domain remove -A com.box.desktop.boxfileprovider)",
+			"(cd /Users/$LOGGED_IN_USER; sudo -u $LOGGED_IN_USER /Applications/Box.app/Contents/MacOS/fpe/streem --remove-fpe-domain-and-archive-unsynced-content Box)",
+			"(cd /Users/$LOGGED_IN_USER; sudo -u $LOGGED_IN_USER /Applications/Box.app/Contents/MacOS/fpe/streem --remove-fpe-domain-and-preserve-unsynced-content Box)",
+			"(cd /Users/$LOGGED_IN_USER; defaults delete com.box.desktop)",
+			"echo \"${LOGGED_IN_USER} ALL = (root) NOPASSWD: /Library/Application\\ Support/Box/uninstall_box_drive_r\" >> /etc/sudoers.d/box_uninstall"
```

Comment:
That's so that an explicit `sudo` is not required to run the `uninstall_box_drive_r` binary? We can't do the explicit `sudo` in the uninstall script itself?

Reply:
That's done because there's a `sudo` in the `/Library/Application Support/Box/uninstall_box_drive` script that the brew artifact runs. The problem is that without that line sudo fails because the user can't be asked for a password.

Here's an excerpt from the script:

```sh
if [ 0 -eq "$user_level_uninstall" ]; then
    # Do the work that requires sudo - this work is placed in a separate script to support
    # developers. On Jenkins passwordless sudo is enabled. But on developer machines we
    # shouldn't require a blanket password-less sudo. This is an issue because developers
    # run Chimp locally and Chimp executes these uninstall scripts and it's often the case
    # that there's no STDIN for sudo to use to get the password. Thus, by placing all the
    # logic that requires sudo in a script called uninstall_box_drive_r... the developers
    # can add an password-exemption for just that script in /etc/sudoers that looks like:
    #     <username> ALL = (root) NOPASSWD: /Library/Application\ Support/Box/uninstall_box_drive_r
    #
    sudo "${0%/*}/"uninstall_box_drive_r $fuse_failure_quits $delete_logs
fi
```

Reply:
It passes in some arguments from the main uninstall script. In the `_r` script, it asks that you don't call it independently